src/collectors/collector_worker.go

```diff
@@ -63,6 +66,7 @@ func (cw *CollectorWorker) RunCollectors() {
 		// In theory, a monitoring application (Check MK) will notice the problem.
 		log.Printf("Failed to write run result: %s\n", err.Error())
 	}
+	return cw.updatedConfigs, cw.diffs
```

Comment:
What about putting the `if (len(cw.diffs) > 0) { utils.GitSendEmail() }` right here, rather than returning, and adding complexity to the Application class?

Reply:
Done.